inbm-lib/inbm_common_lib/utility.py

```diff
@@ -233,8 +233,8 @@ def validate_file_type(path: List[str]) -> None:
     """
     logger.debug("Supported file type prefixes: {0}".format(VALID_MAGIC_FILE_TYPE_PREFIXES))
 
-    tarball_list: List[str] = [x for x in path if (not str(x).endswith('.mender') and tarfile.is_tarfile(x))]
-    extracted_file_list: List[str] = []
+    tarball_list: list[str] = [x for x in path if (not str(x).endswith('.mender') and tarfile.is_tarfile(x))]
```

Comment:
**Suggestion:** Use a generator expression instead of a list comprehension for better memory efficiency when you only need to iterate once. [performance]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    tarball_list: Iterator[str] = (x for x in path if (not str(x).endswith('.mender') and tarfile.is_tarfile(x)))
```